std-bits/table/src/main/java/org/enso/table/data/column/storage/numeric/AbstractLongStorage.java

```diff
@@ -166,13 +207,94 @@ public AbstractLongStorage fillMissingFromPrevious(BoolStorage missingIndicator)
   public abstract AbstractLongStorage widen(IntegerType widerType);
 
   @Override
-  public long get(long index) throws ValueIsNothingException {
-    if (isNothing(index)) {
-      throw new ValueIsNothingException(index);
+  public Storage<Long> applyFilter(BitSet filterMask, int newLength) {
+    var builder =
+        Builder.getForLong(IntegerType.INT_64, newLength, BlackholeProblemAggregator.INSTANCE);
```

Comment:
shouldn't this be `getType()` instead of `INT_64`?

Why does filtering possibly widen the type?

Reply:
Yes it seems that this function was moed from `LongStorage` and there it was preserving the type but here it is no more.

Reply:
a mistake will fix.